Content.Server/Changeling/ChangelingSystem.cs

```diff
@@ -0,0 +1,1114 @@
+        else
+            _popup.PopupEntity(Loc.GetString("changeling-equip-end"), uid, uid);
+
+        PlayMeatySound(uid, comp);
+    }
+    private void OnToggleShield(EntityUid uid, ChangelingComponent comp, ref ToggleOrganicShieldEvent args)
+    {
+        if (!TryUseAbility(uid, comp, args))
+            return;
+
+        if (!TryToggleItem(uid, ShieldPrototype, ref comp.ShieldEntity))
+            return;
+
+        if (comp.ShieldEntity != null)
+            _popup.PopupEntity(Loc.GetString("changeling-shield-start"), uid, uid);
+        else
+            _popup.PopupEntity(Loc.GetString("changeling-hand-transform-end"), uid, uid);
+
+        PlayMeatySound(uid, comp);
+    }
+    private void OnShriekDissonant(EntityUid uid, ChangelingComponent comp, ref ShriekDissonantEvent args)
+    {
+        if (!TryUseAbility(uid, comp, args))
+            return;
+
+        DoScreech(uid, comp);
+
+        var pos = _transform.GetMapCoordinates(uid);
+        var power = comp.ShriekPower;
+        _emp.EmpPulse(pos, power, 5000f, power * 2);
+    }
+    private void OnShriekResonant(EntityUid uid, ChangelingComponent comp, ref ShriekResonantEvent args)
+    {
+        if (!TryUseAbility(uid, comp, args))
+            return;
+
+        DoScreech(uid, comp);
+
+        var power = comp.ShriekPower;
+        _flash.FlashArea(uid, uid, power, power * 2f * 1000f);
+
+        var lookup = _lookup.GetEntitiesInRange(uid, power);
+        var lights = GetEntityQuery<PoweredLightComponent>();
+
+        foreach (var ent in lookup)
+            if (lights.HasComponent(ent))
+                _light.TryDestroyBulb(ent);
+    }
+    private void OnToggleStrainedMuscles(EntityUid uid, ChangelingComponent comp, ref ToggleStrainedMusclesEvent args)
+    {
+        if (!TryUseAbility(uid, comp, args))
+            return;
+
+        ToggleStrainedMuscles(uid, comp);
+    }
+    private void ToggleStrainedMuscles(EntityUid uid, ChangelingComponent comp)
+    {
+        if (!comp.StrainedMusclesActive)
+        {
+            _speed.ChangeBaseSpeed(uid, 125f, 150f, 1f);
+            _popup.PopupEntity(Loc.GetString("changeling-muscles-start"), uid, uid);
+            comp.StrainedMusclesActive = true;
+        }
+        else
+        {
+            _speed.ChangeBaseSpeed(uid, 100f, 100f, 1f);
+            _popup.PopupEntity(Loc.GetString("changeling-muscles-end"), uid, uid);
+            comp.StrainedMusclesActive = false;
+        }
+
+        PlayMeatySound(uid, comp);
+    }
+
+    #endregion
+
+    #region Stings
+
+    private void OnStingBlind(EntityUid uid, ChangelingComponent comp, ref StingBlindEvent args)
+    {
+        if (!TrySting(uid, comp, args))
+            return;
+
+        var target = args.Target;
+        if (!TryComp<BlindableComponent>(target, out var blindable) || blindable.IsBlind)
+            return;
+
+        _blindable.AdjustEyeDamage((target, blindable), 5);
```

Comment:
Isn't it a bit too much? Max eye health is 9

Reply:
didn't know